app/src/main/java/edu/temple/templetag/TagDetailActivity.java

```diff
@@ -83,12 +84,38 @@ protected void onCreate(Bundle savedInstanceState) {
         delBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Toast.makeText(TagDetailActivity.this, "Implement delete function with Firebase", Toast.LENGTH_SHORT).show();
-                /*** TODO Add codes for delete button
-                 *
-                 *
-                 *
-                 * */
+                final String LOG_OUT = "TagDetailActivity";
+                if (mTag.getmTagImageURI().contains("https://firebasestorage.googleapis.com")) {
+                    StorageReference expiredTagImageRef = firebaseStorage.getReferenceFromUrl(mTag.getmTagImageURI());
+                    expiredTagImageRef.delete().addOnSuccessListener(new OnSuccessListener<Void>() {
+                        @Override
+                        public void onSuccess(Void aVoid) {
+                            Log.d(LOG_OUT, "onSuccess: Tag's image deleted from FirebaseStorage");
+                        }
+                    }).addOnFailureListener(new OnFailureListener() {
+                        @Override
+                        public void onFailure(@NonNull Exception e) {
+                            Log.d(LOG_OUT, "onFailure: There was an error deleting the tag's image from FirebaseStorage: " + e);
+                        }
+                    });
+                }
+
+                firestore.collection("Tags")
```

Comment:
Is there anything currently stopping another user from deleting another user's tag?